semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -196,4 +257,257 @@ public static boolean isValidUrl(String url) {
+    for (String path : URL_PATH_ATTRIBUTES) {
+      if (attributeValueMap.get(path) != null) {
+        String s = attributeValueMap.get(path).getValue();
+        if (StringUtils.isNotBlank(s) && s.startsWith(SLASH)) {
+          return getPathFromUrlObject(s).map(str -> removeTrailingSlash(str));
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpMethod(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String method : METHOD_ATTRIBUTES) {
+      if (attributeValueMap.get(method) != null
+          && !StringUtils.isBlank(attributeValueMap.get(method).getValue())) {
+        return Optional.of(attributeValueMap.get(method).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpScheme(Event event) {
+
+    Optional<String> url = getHttpUrlUtil(event);
+    if (url.isPresent() && isAbsoluteUrl(url.get())) {
+      try {
+        return Optional.of(getNormalizedUrl(url.get()).getProtocol());
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
+      }
+    }
+    return getHttpSchemeUtil(event);
+  }
+
+  private static Optional<String> getHttpSchemeUtil(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String scheme : SCHEME_ATTRIBUTES) {
+      if (attributeValueMap.get(scheme) != null
+          && !StringUtils.isEmpty(attributeValueMap.get(scheme).getValue())) {
+        return Optional.of(attributeValueMap.get(scheme).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpUrl(Event event) {
+    Optional<String> url = getHttpUrlUtil(event);
+    if (url.isPresent() && !isAbsoluteUrl(url.get())) {
+      return Optional.empty();
+    }
+    return getHttpUrlUtil(event);
+  }
+
+  //  input url to populateurlparts
+  private static Optional<String> getHttpUrlUtil(Event event) {
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String url : FULL_URL_ATTRIBUTES) {
+      if (attributeValueMap.get(url) != null) {
+        String urlVal = attributeValueMap.get(url).getValue();
+        if (!StringUtils.isBlank(urlVal) && isValidUrl(urlVal)) {
+          if (isAbsoluteUrl(urlVal)) {
+            return Optional.of(urlVal);
+          } else {
+            Optional<String> url2 = getHttpUrlForOTelFormat(attributeValueMap);
```

Comment:
nit: rename var `url2`

Reply:
to ?

Reply:
something more representative of the value it is holding, I would call it `ulrOtelFormat` maybe